src/extension/infer/test.rs

```diff
@@ -154,36 +155,25 @@ fn plus() -> Result<(), InferExtensionError> {
 }
 
 #[test]
-// This generates a solution that causes validation to fail
```

Comment:
This comment was incorrect (it doesn't generate a solution - the error is CantInfer). But the body/subject of this test has largely switched with the identically-named test in validation.